.cirrus.yml

```diff
@@ -55,8 +55,24 @@ task:
       always:
         format_script: ./script/tool_runner.sh format --fail-on-change
         license_script: dart pub global run flutter_plugin_tools license-check
-        analyze_script: ./script/tool_runner.sh analyze --custom-analysis=web_benchmarks/testing/test_app,flutter_lints/example,rfw/example,metrics_center
+        analyze_script: ./script/tool_runner.sh analyze --custom-analysis=script/configs/custom_analysis.yaml
         pubspec_script: ./script/tool_runner.sh pubspec-check
+    # Does a sanity check that packages at least pass analysis on the N-1 and N-2
+    # versions of Flutter stable if the package claims to support that version.
+    # This is to minimize accidentally making changes that break old versions
+    # (which we don't commit to supporting, but don't want to actively break)
+    # without updating the constraints.
+    # Note: The versions below should be manually updated after a new stable
+    # version comes out.
+    - name: legacy-version-analyze
+      depends_on: format+analyze
+      env:
+        matrix:
+          CHANNEL: "2.5.3"
+          CHANNEL: "2.8.1"
+      analyze_script:
+        - ./script/tool_runner.sh analyze --skip-if-not-supporting-flutter-version="$CHANNEL" --custom-analysis=script/configs/custom_analysis.yaml
```

Comment:
Will the echo actually print out if the analyze fails?  I would expect the exit code to be non-zero which would short circuit everything.

Reply:
🤦🏻 

I'll just remove this; the contributing docs specifically cover this already, so it was just a nice-to-have. (I don't see a good way to add it; if I move it to an "always" step it'll have to be a separate step in the UI, which would be collapsed by default when the earlier step failed, which means nobody would see that either, so it would require a special tool flag or something to output which is too much complexity for the value.)

Reply:
could you do `if ! ./script/tool_runner.sh analyze; then echo "foo"; exit 1; fi`?

Reply:
Ah, that would probably work. I'll land without it, and revisit if it ends up causing confusion in pratice.